app/controllers/pieces_controller.rb

```diff
@@ -101,7 +101,6 @@ def game_has_two_players?(game)
 
   def check_test(piece, x, y)
     return false if piece.can_take?(helpers.get_piece(x, y, piece.game)) && piece.type == 'King'
```

Comment:
As I suggested elsewhere just as I am reviewing this popped up for me so figured should at least ask about it.
`piece.can_take?(helpers.get_piece(x, y, piece.game))` This should always return true since this would happen after the `valid_move?` check in the update action.
`piece.type == 'King' checks if the piece being moved is a king? (or perhaps if the piece being taken is king but pretty sure this is testing if piece being moved is a king.
This seems to suggest if you tried you could move a king into check since this would return false before the if statement on line 105/106 returned false with a flash message that prevents the `move_to!` method from happening on line 14 of pieces_controller.rb.

Reply:
Yes it is checking if the piece being moved is a king. 
This method would not allow a king to take a piece that would then put it into check due to the next line of code:

```
    if piece.puts_self_in_check?(x, y)
      flash.now[:alert] << 'You cannot put/leave yourself in Check.'
```

Reply:
`valid_move?` for each piece model does not check if it is an enemy piece on the current tile. `can_take?` does.

Reply:
`valid_move?` checks if your own piece is on the destination tile which is the exact same as checking if you can take a piece on the destination tile. Refer to line 11 of piece.rb `if players_own_piece_is_there?(x, y)`.

Reply:
Since 103 is a return line 105 that checks for putting in check would never be reached.

Reply:
Ahh yea, you are right about the return on line 103. Not sure why I wasn't thinking about that. I will update the PR and switch those lines.